app/controllers/api/tags_controller.rb

```diff
@@ -0,0 +1,85 @@
+module Api
+  # Api controller for tags
+  # Uses Rails' RESTful routes (check 'rake routes' for the configured routes)
+  class TagsController < MainApiController
+    DEFAULT_FIELDS = [:id, :name, :description, :creator, :use].freeze
+
+    def index
+      @assignment = Assignment.find_by(id: params[:assignment_id])
+
+      respond_to do |format|
+        parent = @assignment || current_course
+        tags = parent.tags.includes(:role, :groupings).order(:name)
+
+        tag_info = tags.map do |tag|
+          {
+            id: tag.id,
+            name: tag.name,
+            description: tag.description,
+            creator: tag.role.display_name,
+            use: tag.groupings.size
+          }
+        end
+        format.xml do
+          render xml: tag_info
+        end
+        format.json do
+          render json: tag_info
+        end
+      end
+    end
+
+    def edit
+      @tag = Tag.find(params[:id])
+    end
+
+    # Creates a new instance of the tag.
+    def create
+      params.permit(:name, :description)
+      begin
+        new_tag = Tag.new(name: params[:name], description: params[:description],
+                          course: Course.find_by(id: params[:course_id]),
```

Comment:
One of the modules we include in all controllers is `SessionHandler`, which defines a `current_course` helper method you should call here